### ISSUE TITLE
Comment pagination should use a nav and an accessible name

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#63328]
### Description

The comment pagination blocks are navigation items and should be placed inside a `<nav>` element to be more accessible. Currently a `<div>` is used.

The navigation should have a name that helps users separate the comment navigation from the main navigation blocks,
post navigation, query navigation etc. This can be an aria label: "Comments".

The WordPress functions that are the classic theme equivalents use a `<nav>` by default, and have options both for screen reader text and aria label:
https://developer.wordpress.org/reference/functions/get_the_comments_navigation/
https://developer.wordpress.org/reference/functions/get_the_comments_pagination/

### Step-by-step reproduction instructions

Add a comment pagination block.
Confirm that it uses a div.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by youknowriad** [comment]
I think this was fixed in #70730
